DigitalLearningSolutions.Web.Tests/Helpers/RegistrationEmailValidatorTests.cs

```diff
@@ -351,6 +358,220 @@ string centreSpecificEmail
         }
 
+        [Test]
+        public void
+            ValidateEmailsForInternalAdminRegistration_adds_no_new_error_messages_when_centre_email_is_already_invalid()
+        {
+            // Given
+            var model = GetDefaultAdminInformationViewModel(centreSpecificEmail: null);
+
+            modelState.AddModelError(nameof(InternalAdminInformationViewModel.CentreSpecificEmail), DefaultErrorMessage);
+
+            // When
+            RegistrationEmailValidator.ValidateEmailsForInternalAdminRegistration(
+                DefaultUserId,
+                model,
+                modelState,
+                userDataService,
+                centresService
+            );
+
+            // Then
+            A.CallTo(
+                () => userDataService.CentreSpecificEmailIsInUseAtCentre(A<string>._, A<int>._, A<IDbTransaction?>._)
+            ).MustNotHaveHappened();
+            modelState[nameof(PersonalInformationViewModel.CentreSpecificEmail)].ValidationState.Should()
+                .Be(ModelValidationState.Invalid);
+            AssertModelStateErrorIsExpected(
+                nameof(PersonalInformationViewModel.CentreSpecificEmail),
+                DefaultErrorMessage
+            );
+        }
+
+        [Test]
+        public void ValidateEmailsForInternalAdminRegistration_adds_error_on_centre_email_if_already_in_use()
+        {
+            // Given
+            var model = GetDefaultAdminInformationViewModel();
+
+            A.CallTo(
+                () => userDataService.CentreSpecificEmailIsInUseAtCentre(
+                    DefaultCentreSpecificEmail,
+                    DefaultCentreId,
+                    A<IDbTransaction?>._
+                )
+            ).Returns(true);
+            A.CallTo(() => userDataService.GetCentreEmail(DefaultUserId, DefaultCentreId))
+                .Returns("another.centre@email");
+
+            // When
+            RegistrationEmailValidator.ValidateEmailsForInternalAdminRegistration(
+                DefaultUserId,
+                model,
+                modelState,
+                userDataService,
+                centresService
+            );
+
+            // Then
+            A.CallTo(
+                () => userDataService.CentreSpecificEmailIsInUseAtCentre(
+                    DefaultCentreSpecificEmail,
+                    DefaultCentreId,
+                    A<IDbTransaction?>._
+                )
+            ).MustHaveHappenedOnceExactly();
+            modelState[nameof(PersonalInformationViewModel.CentreSpecificEmail)].ValidationState.Should()
+                .Be(ModelValidationState.Invalid);
+            AssertModelStateErrorIsExpected(
+                nameof(PersonalInformationViewModel.CentreSpecificEmail),
+                DuplicateEmailErrorMessage
+            );
+        }
+
+        [Test]
+        public void
+            ValidateEmailsForInternalAdminRegistration_adds_wrong_email_error_on_centre_email_if_neither_email_not_matches_centre()

```

Comment:
You haven't covered all the cases here. You need tests for:
* primary email matches centre + centre specific email is null
* primary email does not match centre + centre specific email is null
* primary email matches centre + centre specific email matches centre
* primary email matches centre + centre specific email does not match centre
* primary email does not match centre + centre specific email matches centre
* primary email does not match centre + centre specific email does not match centre

You have covered some of these cases, but I think it might be easier to see which if they were present in a test with `TestCase`s, which does either the `AssertModelStateErrorIsExpected` check, or the `modelState.ValidationState.Should().Be(ModelValidationState.Valid);` depending on expected output.

(I think that that is more important than checking how many times `DoesEmailMatchCentre` is called, which would be more fiddly to get right in a singe test with `TestCase`s)